### DISCUSSION TITLE
Missing Storage Manager Callbacks Since 0.19.4

### DISCUSSION THREAD

**Discussion by at-codes** [GrapesJS/grapesjs#4826] [Q&A] [answered]
Storage manager callbacks were added because of [this](https://github.com/artf/grapesjs/issues/876) issue. This [commit](https://github.com/artf/grapesjs/commit/b80e1eca26ef661e20aa1fd5674e583c5c0e485e) removed the changes. I need this functionality to solve my issue. I see that in v0.19.4 this functionality was removed. Why was it removed without a replacement? It's cool it's an async function now, but we lost functionality to get any response back from the server. Has anyone been able to work around this issue? Do I need to do some kind of override or second POST to properly get the response back? Do I need to write a PR to get this functionality back?

**Comment by artf** [comment, marked as answer]
As store/load functions are now async you don't need callbacks anymore, just get it from the promise.

**Reply by at-codes** [reply]
```javascript
    editor.store().then((r) => {
        console.log(r);        
    });
```
Doing something like the above only returns the project data ie assets, pages, styles. I can see what I want in the network devtools post response but not in the console log above. 

Maybe the issue is that [here](https://github.com/artf/grapesjs/blob/master/src/storage_manager/model/RemoteStorage.js#L6) there is no return. That would explain why I'm always only seeing the project data [here](https://github.com/artf/grapesjs/blob/master/src/editor/model/Editor.ts#L749). If I'm following the code correctly

**Reply by artf** [reply]
Oh yeah in that case `editor.store` doesn't track the response from the `Storage.store` call it self as most of the time it might be empty or different for each integration. I think one way to fix it would be to track the real response and include it with the `storage:store` event but for the `editor.store` call we should always have a predictable output.

**Reply by at-codes** [reply]
It would be very convenient to get [this](https://github.com/artf/grapesjs/blob/master/src/editor/model/Editor.ts#L727) response when you do need data from it. I can see the load function below it returns the response.
I have tried the `storage:store` event and variations of it. That still only gives the project data.
We were able to do a workaround to fill our use case.

**Reply by artf** [reply]
I'll try then to add the actual response to the `storage:store` event but for now there is no way to get it.

**Reply by artf** [reply]
https://github.com/artf/grapesjs/commit/e6a086afac77d8da73f225dd0aa84b3d178ec47b

**Reply by at-codes** [reply]
Thank you!
